### ISSUE TITLE
Fullscreen doesn't work properly on Windows 7 & 10

### ISSUE THREAD

**Issue by a-amouei** [qTox/qTox#5782]
##### Brief Description

OS: Windows 7 Ultimate, Service Pack 1
qTox version: v1.16.3.
Commit hash:  8eed684c37bd711de2ca4bd863f5f0509edcdc2d
toxcore:  0.2.3
Qt:  5.9.6

Reproducible: Always (every time that we tried)

##### Steps to reproduce

1. click on the fullscreen button

##### Observed Behavior

"When I clicked on 'Fullscreen', the usually inlying screen was transferred to the top left of my windows screen without filling it. The size stayed the same without becoming a fullscreen. Furthermore, there were no buttons to turn back to the normal screen (inlay). Also shortcuts like 'Esc' or 'Alt + F4' did not work. It only stopped when the call ended."

This description is from a colleague who faced this problem (with qTox v1.16.3 on Windows 10) when we were talking through qTox. If it's needed, I can ask him for more details.

UPDATE: Today I checked it on my laptop with Windows 7 and exactly the same happened. So I'm going to provide details from my own system. Also updated the "Brief Description" section above to describe my system.

**Comment by sudden6** [comment]
Can you please attach the following details:

- qTox log output
- number of screens attached
- resolution of the screens

**Comment by a-amouei** [comment]
The same occurred on my Windows 7 system when testing with "Tox EchoBot", so I'm going to provide the additional information from my own system (I also updated the initial post accordingly).

There is only one screen (= the display of my laptop)
Resolution: 1366 x 768
Please find attached a screenshot and the debug log file.

![screenshot](https://user-images.githubusercontent.com/48631976/62942746-30329c00-bdee-11e9-8aac-0b5cf9049d12.jpg)
[debuglog.txt](https://github.com/qTox/qTox/files/3497020/debuglog.txt)

It also seems that the microphone doesn't work on Windows. But I first should check the Windows settings.

Please feel free to ask for any further information.

**Comment by sudden6** [comment]
Hmm, the log doesn't show anything interesting.

Can you test with a nightly build: https://github.com/qTox/qTox-nightly-releases/releases/tag/ci-master-latest?

**Comment by a-amouei** [comment]
I installed setup-qtox-x86_64-release.exe from the URL you mentioned and tested. The bug still exists.

New debug log: [debuglog.txt](https://github.com/qTox/qTox/files/3497196/debuglog.txt)

I'm ready for doing any further tests.

**Comment by sudden6** [comment]
Ok, that at least confirms the problem is still there in master. I just tested this on Linux and it doesn't seem to happen there.

There seems to be no short term solution until someone decides to debug this issue on Windows.

@anthonybilinski do you think we should add this to the v1.17 milestone?

**Comment by a-amouei** [comment]
I don't face it on Linux (Ubuntu 18.04) either.

**Comment by anthonybilinski** [comment]
@sudden6 I think so. The bug is very likely within these few lines https://github.com/qTox/qTox/blob/412f14558150e4da76acb3d639f5a23dca58b33b/src/video/genericnetcamview.cpp#L134, and might be solvable with a little reading of Qt docs and investigation.

**Comment by ghost** [comment]
I have been having exactly the same issue on my Windows 7 SP1 x64.
In qTox v.1.14.1, v1.16.3, the today's nightly, and 2-3 more nightlies i've tested some time earlier.

**Comment by GoFreeman** [comment]
Got the same on all my PCs (Windows 7 64-bit) and all the PCs of my friends (Win7-64 and Win10-64).
With version v1.16.3 as well as the nightly versions.
The only issue that made us temporarily choose another Skype alternative for now and stopping from finally moving on to qTox. :-/

**Comment by anthonybilinski** [comment]
My previous comment saying that this issue is likely trivial was wrong, some investigation has gone into it but it's not clear what we're doing wrong. Even though this is a serious issue, since it's not a regression I'm going to kick it out of the [v1.17.0 milestone](https://github.com/qTox/qTox/milestone/23) to unblock the release so that we can get many other fixes released. I'll bump the priority of this due to it being a fairly core feature and effecting multiple users.

**Comment by ghost** [comment]
I would like to ask to include this issue in the v1.17 milestone if it is possible,
not only because it is a serious bug,
but also because this issue is clearly visible during the use and can make bad first impression on new users who try out qTox and consider choosing it over the other alternatives. :-)

**Comment by anthonybilinski** [comment]
I'll take another look at it and see if I can make some progress.

**Comment by e14tech** [comment]
I’d like to add that I’m having this same issue on Windows 10 1909.

Works perfectly fine on Ubuntu 18.04 LTS.

Downloaded the latest release build today, 64 bit.

**Comment by TorstenFunk** [comment]
I have the same issue with the nightly build from 23th March. My chat partner used a webcam (Logitech 220) set to 480p on Win10. I used a webcam (Logitech  C170) set to 480p on Win7.
After switching to full-screen, the video frame was moved into top-left corner of my screen, not scaled up to cover entire HD screen (it remained at 480p), and there was no button overlay over the video, e.g. to end full-screen. Only way to end this broken full-screen mode was to end the call.

**Comment by ghost** [comment]
I see there is a new v1.18 milestone, could this issue be included in it?

**Comment by spiralofhope** [comment]
I also have this problem under the most recent release:

- Windows 10
- qTox 1.17.2
- commit hash `465d82606813439ee14425f7462903bb39f2da30`
- toxcore 0.2.11
- Qt version 5.12.8

A workaround has been to use AltDrag:

https://stefansundin.github.io/altdrag/

I `click-drag` to resize the video window to nearly fullscreen (my taskbar is overtop) and then that window shows its controls at the bottom, allowing me to the click fullscreen icon.

**Comment by yekemezek** [comment]
Exactly the same issue as everyone else.

- Windows 7 Ultimate x64
- qTox 1.17.3
- toxcore 0.2.12
- Qt version: 5.12.9

**Comment by adguardgirl** [comment]
Exactly the same issue. 


    Windows 10 pro 64
    qTox 1.17.3
    toxcore 0.2.12
    Qt version: 5.12.9

**Comment by ghost** [comment]
The same issue.

- Windows 10 Home Edition 64-bit
- qTox v1.17.3
- Commit hash: c0e9a3b79609681e5b9f6bbf8f9a36cb1993dc5f
- Toxcore 0.2.12
- Qt 5.12.9

**Comment by ghost** [comment]
persists in 1.17.6

**Comment by NthAdept** [comment]
I'm having this issue with Win10, using qTox v1.17.4.  I don't seem to be able to update to v1.17.6.

**Comment by anthonybilinski** [comment]
To upgrade you should download the installer from https://github.com/qTox/qTox/releases/tag/v1.17.6 i.e. [setup-qtox-x86_64-release.exe](https://github.com/qTox/qTox/releases/download/v1.17.6/setup-qtox-x86_64-release.exe), close the running qTox, then run the installer. If that's giving you errors, please open a new issue since it's unrelated to fullscreen functionality.

**Comment by NthAdept** [comment]
Thank you very much, anthonybilinski!  Your instructions for upgrading to v1.17.6 worked flawlessly.  I suspect that I was NOT shutting down the running qTox PRIOR to running the installer for the new version.  I will follow your procedure going forward.  
My full screen issue was solved EVER so simply when a qTox pal FINALLY came along and asked if I had tried F11.  I tried this and it instantly fixed me up.  I can toggle in and out of full screen mode at will.  I must have inadvertently toggled this into full screen mode.  This is one of an infinite number of "basic" computer processes that I had not learned until now.  Thank you sir, for your quick response to my issue.  qTox is INVALUABLE to me, so it is MOST excellent to have learned these lessons.
